dtfs-central-api/src/services/portal/deal.service.ts

```diff
@@ -47,28 +47,35 @@ export class PortalDealService {
    * @param addGefDealCancelledActivityParams
    * @param addGefDealCancelledActivityParams.dealId - the deal
    * @param addGefDealCancelledActivityParams.author - the activity's author
+   * @param addGefDealCancelledActivityParams.cancellationIsInFuture - cancellation is in the future
    * @param addGefDealCancelledActivityParams.auditDetails - the users audit details
    */
   public static async addGefDealCancelledActivity({
     deal,
     author,
+    cancellationIsInFuture,
     auditDetails,
   }: {
     deal: TfmDeal;
     author: ActivityAuthor;
+    cancellationIsInFuture?: boolean;
     auditDetails: AuditDetails;
   }): Promise<void> {
     if (deal.dealSnapshot.dealType === DEAL_TYPE.GEF) {
       const { _id: dealId } = deal.dealSnapshot;
 
+      const type = cancellationIsInFuture ? PORTAL_ACTIVITY_TYPE.DEAL_CANCELLATION_PENDING : PORTAL_ACTIVITY_TYPE.DEAL_CANCELLED;
+
+      const label = cancellationIsInFuture ? PORTAL_ACTIVITY_LABEL.DEAL_CANCELLATION_PENDING : PORTAL_ACTIVITY_LABEL.DEAL_CANCELLED;
```

Comment:
since these 2 lines are the same, could `cancellationIsInFuture ? PORTAL_ACTIVITY_LABEL.DEAL_CANCELLATION_PENDING : PORTAL_ACTIVITY_LABEL.DEAL_CANCELLED;` be a function?

Reply:
They're not the same, one uses `PORTAL_ACTIVITY_TYPE`, the other uses `PORTAL_ACTIVITY_LABEL`.

Reply:
This is now outdated - resolving